drivers/volume/portworx/portworx.go

```diff
@@ -261,42 +300,202 @@ func (p *portworx) getSnapshotName(tags *map[string]string) string {
+		request := &api.CloudBackupCreateRequest{
+			VolumeID:       volumeID,
+			CredentialUUID: snapshotCredID,
+		}
+		err := p.volDriver.CloudBackupCreate(request)
+		if err != nil {
+			return nil, getErrorSnapshotConditions(err), err
+		}
+
+		logrus.Infof("cloudsnap backup started successfully. waiting for completion...")
+		err = wait.ExponentialBackoff(cloudsnapBackoff, func() (bool, error) {
+			snapshotID, err = p.checkCloudSnapStatus(volumeID)
+			if err != nil {
+				logrus.Error(err)
+				return false, nil
+			}
+
+			return true, nil
+		})
+		if err != nil {
+			return nil, getErrorSnapshotConditions(err), err
+		}
+	case crdv1.PortworxSnapshotTypeLocal:
+		if isGroupSnap(snap) {
+			groupID := snap.Metadata.Annotations[pxSnapshotGroupIDKey]
+			groupLabels := parseGroupLabelsFromAnnotations(snap.Metadata.Annotations)
+
+			infoStr := fmt.Sprintf("creating group snapshot: [%s] %s", snap.Metadata.Namespace, snap.Metadata.Name)
+			if len(groupID) > 0 {
+				infoStr = fmt.Sprintf("%s. Group ID: %s", infoStr, groupID)
+			}
+
+			if len(groupLabels) > 0 {
+				infoStr = fmt.Sprintf("%s. Group labels: %v", infoStr, groupLabels)
+			}
+
+			logrus.Infof(infoStr)
+
+			resp, err := p.volDriver.SnapshotGroup(groupID, groupLabels)
+			if err != nil {
+				return nil, getErrorSnapshotConditions(err), err
+			}
+
+			if len(resp.Snapshots) == 0 {
+				err = fmt.Errorf("found 0 snapshots using given group selector/ID")
+				return nil, getErrorSnapshotConditions(err), err
+			}
+
+			snapIDs := make([]string, 0)
+			snapDataNames := make([]string, 0)
+			failedSnapVols := make([]string, 0)
+
+			// Loop through the response and check if all succeeded. Don't create any k8s objects if
+			// any of the snapshots failed
+			for volID, newSnapResp := range resp.Snapshots {
+				if newSnapResp.GetVolumeCreateResponse() == nil {
+					err = fmt.Errorf("Portworx API returned empty response on creation of group snapshot")
+					return nil, getErrorSnapshotConditions(err), err
+				}
+
+				if newSnapResp.GetVolumeCreateResponse().GetVolumeResponse() != nil &&
+					len(newSnapResp.GetVolumeCreateResponse().GetVolumeResponse().GetError()) != 0 {
+					logrus.Errorf("Failed to create snapshot for volume: %s due to: %v",
+						volID, newSnapResp.GetVolumeCreateResponse().GetVolumeResponse().GetError())
+					failedSnapVols = append(failedSnapVols, volID)
+					continue
+				}
+			}
+
+			if len(failedSnapVols) > 0 {
+				err = fmt.Errorf("all snapshots in the group did not succeed. failed: %v succeeded: %v",
+					failedSnapVols, snapIDs)
+				return nil, getErrorSnapshotConditions(err), err
+			}
+
+			for _, newSnapResp := range resp.Snapshots {
+				newSnapID := newSnapResp.GetVolumeCreateResponse().GetId()
+				snapObj, err := p.createVolumeSnapshotCRD(newSnapID, newSnapResp, groupID, groupLabels, snap)
+				if err != nil {
+					err = fmt.Errorf("failed to create CRD for PX snapshot: %s due to: %v", newSnapID, err)
```

Comment:
This will leave snaps for some of them. Either need to delete all of them or try creating the rest too.